Core/include/Acts/SpacePointFormation/SpacePointBuilder.hpp

```diff
@@ -0,0 +1,160 @@
+  /// @param stripEndsPair The global positions of the ends of the strip pair
+  /// @param spacePoint The space point to be created.
+  /// @note If no configuration is set, the default values will be used
+  // template <template <typename...> typename container_t>
+  void calculateDoubleHitSpacePoint(
+      const Acts::GeometryContext& gctx,
+      const std::pair<const Measurement*, const Measurement*>& measurementPair,
+      const std::pair<const std::pair<Vector3, Vector3>,
+                      const std::pair<Vector3, Vector3>>& stripEndsPair,
+      spacepoint_t* spacePoint) const;
+
+ protected:
+  /// @brief Getter method for the local coordinates of a measurement
+  /// on its corresponding surface
+  ///
+  /// @param meas measurement that holds the neccesary information of the hit position.
+  /// @return vector of the local coordinates of the measurement on the surface
+  Vector2 getLocalPos(const Measurement& meas) const;
+  std::pair<Acts::Vector2, Acts::SymMatrix2> getLocalPosCov(
+      const Measurement& meas) const;
+
+  /// @brief Getter method for the global coordinates of a measurement
+  ///
+  /// @param gctx The current geometry context object, e.g. alignment
+  /// @param meas measurement that holds the necessary
+  /// information
+  /// @return vectors of the global coordinates and covariance of the measurement
+  std::pair<Vector3, Vector2> globalCoords(const GeometryContext& gctx,
+                                           const Measurement& meas) const;
+
+  /// @brief Get global covariance from the local position and covariance
+  /// @param gctx The current geometry context object, e.g. alignment
+  /// @param geoId The geometry ID
+  /// @param localPos The local position
+  /// @param localCov The local covariance matrix
+  /// @return (rho, z) components of the global covariance
+  Acts::Vector2 globalCov(const Acts::GeometryContext& gctx,
+                          const Acts::GeometryIdentifier& geoId,
+                          const Acts::Vector2& localPos,
+                          const Acts::SymMatrix2& localCov) const;
+
+  /// @brief Get the first component of the local covariance.
+  /// @param meas The measurement
+  /// @return the (0, 0) component of the local covariance
+  double getLoc0Var(const Measurement& meas) const;
+
+  /// @brief Calculate the global covariance from the front and back measurement in the strip SP formation
+  /// @param gctx The current geometry context object, e.g. alignment
+  /// @param measFront The measurement on the front layer
+  /// @param measBack The measurement on the back layer
+  /// @param theta The angle between the two strips
+  /// @return (rho, z) components of the global covariance
+  Acts::Vector2 calcGlobalVars(const Acts::GeometryContext& gctx,
+                               const Measurement& measFront,
+                               const Measurement& measBack,
+                               const double theta) const;
+
```

Comment:
I would suggest that this sort of utility function is part of a utility class and not the builder.

Reply:
I'm moving these functions to `Utilities/SpacePointUtility.hpp`